src/api/layers.api.js

```diff
@@ -135,46 +134,150 @@ export class WMTSLayer extends Layer {
+    }
+
+    getURL() {
+        return this.geoJsonUrl;
     }
 }
 
-const generateClassForLayerConfig = (layerConfig) => {
+/**
+ * A sub-layer of an aggregate layer. Will define at which resolution this sub-layer should be shown (shouldn't overlap other sub-layers from the aggregate)
+ */
+export class AggregateSubLayer {
+    /**
+     * @param {String} subLayerId the ID used in the BOD to describe this sub-layer
+     * @param {Layer} layer the sub-layer config (can be a {@link GeoJsonLayer}, a {@link WMTSLayer} or a {@link WMTSLayer})
+     * @param {Number} minResolution in meter/px, at which resolution this sub-layer should start to be visible
+     * @param {Number} maxResolution in meter/px, from which resolution the layer should be hidden
+     */
+    constructor(subLayerId, layer, minResolution = Number.MIN_SAFE_INTEGER, maxResolution = Number.MAX_SAFE_INTEGER) {
+        this.subLayerId = subLayerId;
+        this.layer = layer;
+        this.minResolution = minResolution;
+        this.maxResolution = maxResolution;
+    }
+}
+
+/**
+ * An aggregate layer is a combination of 2 or more layers where only one of them will be shown at a time.
+ * Which one is shown is decided by the map resolution, and by the min/max resolution of all sub-layer's config
+ */
+export class AggregateLayer extends Layer {
+    /**
+     * @param {String} name the name of this layer in the given lang
+     * @param {String} id the layer ID in the BOD
+     * @param {Number} opacity the opacity to be applied to this layer
+     * @param {TimeConfig} timeConfig time series config (if available)
+     */
+    constructor(name, id, opacity, timeConfig) {
+        super(name, LayerTypes.AGGREGATE, id, opacity);
+        this.timeConfig = timeConfig;
+        this.subLayers = [];
+    }
+
+    /**
+     * @param {AggregateSubLayer} subLayer
+     */
+    addSubLayer(subLayer) {
+        this.subLayers.push(subLayer);
+    }
+
+    getURL() {
+        throw new Error("Aggregate layers shouldn't be asked directly for URL, but sub-layers should")
+    }
+}
+
+const generateClassForLayerConfig = (layerConfig, id, allOtherLayers, lang) => {
     let layer = undefined;
     if (layerConfig) {
-        const { serverLayerName: id, label: name, type, opacity, format, background } = layerConfig;
-        let timeEnableConfig = null;
-        if (layerConfig.timeEnabled) {
-            timeEnableConfig = new TimeSeriesConfig(layerConfig.timeBehaviour, layerConfig.timestamps);
-        }
+        const { label: name, type, opacity, format, background } = layerConfig;
+        const timeConfig = new TimeConfig(layerConfig.timeBehaviour, layerConfig.timestamps);
         switch(type.toLowerCase()) {
             case 'wmts':
-                layer = new WMTSLayer(name, id, opacity, format, timeEnableConfig, background, WMTS_BASE_URL);
+                layer = new WMTSLayer(name, id, opacity, format, timeConfig, !!background, WMTS_BASE_URL);
```

Comment:
Is it a typo or does have `!!` a special meaning in javascript that I'm not aware of ?

Reply:
type forcing to `boolean` ;-) little JS trick